plugin.video.catchuptvandmore/resources/lib/addon_utils.py

```diff
@@ -25,43 +25,72 @@
 # It makes string literals as unicode like in Python 3
 from __future__ import unicode_literals
 
+# Core imports
 import os
-import sys
-import pickle
-import binascii
 
+# Kodi imports
 from kodi_six import xbmcgui
 from kodi_six import xbmc
 from kodi_six import xbmcvfs
+from codequick import Script, utils
+import urlquick
 
-try:
-    from urllib.parse import urlunsplit
-except ImportError:
-    from urlparse import urlunsplit
 
-import urlquick
-from codequick import Script
+# Local imports
 from resources.lib.labels import LABELS
 
 
-PY3 = sys.version_info[0] >= 3
+def get_item_label(item_id):
+    """Get (translated) label of 'item_id'
+
+    Args:
+        item_id (str)
+    Returns:
+        str: (translated) label of 'item_id'
+    """
+    label = item_id
+    if item_id in LABELS:
+        label = LABELS[item_id]
+        if isinstance(label, int):
+            label = Script.localize(label)
+    return label
+
+
+def get_item_media_path(item_media_path):
+    """Get full path or URL of an item_media
+
+    Args:
+        item_media_path (str or list): Partial media path of the item (e.g. channels/fr/tf1.png)
+    Returns:
+        str: Full path or URL of the item_pedia
+    """
+    full_path = ''
 
+    # Local image in ressources/media folder
+    if type(item_media_path) is list:
+        full_path = os.path.join(Script.get_info("path"), "resources", "media",
+                                 *(item_media_path))
 
-def build_kodi_url(route_path, raw_params):
-    # route_path: /resources/lib/channels/fr/mytf1/get_video_url/
-    # raw_params: params dict
-    if raw_params:
-        pickled = binascii.hexlify(
-            pickle.dumps(raw_params, protocol=pickle.HIGHEST_PROTOCOL))
-        query = "_pickle_={}".format(
-            pickled.decode("ascii") if PY3 else pickled)
+    # Remote image with complete URL
+    elif 'http' in item_media_path:
+        full_path = item_media_path
 
-    # Build kodi url
-    return urlunsplit(
-        ("plugin", "plugin.video.catchuptvandmore", route_path, query, ""))
+    # Remote image on our images repo
+    else:
+        full_path = 'https://github.com/Catch-up-TV-and-More/images/raw/master/' + item_media_path
```

Comment:
Thank you enen92 for your review 👍 
I understand your point of view concerning external resources.
A while ago we made the decision to move all channels logo images in a dedicated GitHub repo because:
* We also use channel logos on our website, so each channel logo is only on one place and both the website and our add-on can access the logo file
* In the past all channel logos were bundle in the add-on. But now we support a lot of channels in our add-on, so with all the images and fanarts the size of our add-on seemed too big for us, this is why we moved images on GitHub.

I will discuss with the co-developer of Catch-Up TV & More in order to find a solution to this 😉

Reply:
If the resource addon is included in our repo you can then just link to the image stored in github. You still have full control over the images displayed. Resoruces/images will then be managed as any other addon

Reply:
Done